core/src/main/java/org/cru/contentscoring/core/servlets/CopyScoresToTagsServlet.java

```diff
@@ -0,0 +1,259 @@
+    private void moveScoresToTags(
+        final List<Hit> results,
+        final TagManager tagManager,
+        final ResourceResolver resourceResolver) throws RepositoryException, ReplicationException {
+
+        List<Resource> pages = Lists.newArrayList();
+        for (Hit result : results) {
+            Resource page = result.getResource();
+            copyScoreTagToPrimaryExperienceFragment(page, resourceResolver, tagManager);
+            moveScoreToTag(page, tagManager);
+            pages.add(page);
+        }
+
+        replicatePages(pages, resourceResolver);
+    }
+
+    private void moveScoreToTag(final Resource page, final TagManager tagManager) throws RepositoryException {
+        Resource pageContent = getJcrContent(page);
+
+        if (pageContent != null) {
+            String score = pageContent.getValueMap().get("score", String.class);
+            pageContent.adaptTo(Node.class).getProperty("score").remove();
+
+            Set<Tag> tags = buildTagsWithScore(pageContent, tagManager, score);
+            tagManager.setTags(pageContent, tags.toArray(new Tag[0]));
+        }
+    }
+
+    private void copyScoreTagToPrimaryExperienceFragment(
+        final Resource page,
+        final ResourceResolver resourceResolver,
+        final TagManager tagManager) {
+
+        Resource pageContent = getJcrContent(page);
+
+        if (pageContent != null) {
+            String score = pageContent.getValueMap().get("score", String.class);
+            String primaryExperienceFragmentPath = pageContent.getValueMap().get(PRIMARY_XF_NAME, String.class);
+
+            if (!Strings.isNullOrEmpty(primaryExperienceFragmentPath)) {
+                Resource primaryExperienceFragment = resourceResolver.getResource(primaryExperienceFragmentPath);
+                Resource experienceFragment;
+
+                if (primaryExperienceFragment != null) {
+                    if (isExperienceFragmentVariation(primaryExperienceFragment)) {
+                        experienceFragment = primaryExperienceFragment.getParent();
+                    } else if (isExperienceFragment(primaryExperienceFragment)) {
+                        experienceFragment = primaryExperienceFragment;
+                    } else {
+                        return;
+                    }
+
+                    Resource experienceFragmentContent = getJcrContent(experienceFragment);
+                    Set<Tag> tags = buildTagsWithScore(experienceFragmentContent, tagManager, score);
+                    tagManager.setTags(experienceFragmentContent, tags.toArray(new Tag[0]));
+
+                    for (Resource child : experienceFragment.getChildren()) {
+                        Resource childContent = getJcrContent(child);
+                        if (childContent != null) {
+                            Set<Tag> childTags = buildTagsWithScore(childContent, tagManager, score);
+                            tagManager.setTags(childContent, childTags.toArray(new Tag[0]));
```

Comment:
These two lines are essentially repeated 3 times; I'd consider extracting them to a method